Builds/CMake/CMakeFuncs.cmake

```diff
@@ -105,16 +105,36 @@ macro(parse_target)
       endif()
 
     endwhile()
+    set(CMAKE_C_COMPILER ${CMAKE_C_COMPILER} CACHE FILEPATH
```

Comment:
@bachase cache variables and "regular" variables are distinct. We can have a cache variable `foo` with value `true` and a regular variable with value `false`. The `unset` above removes the "regular" variable to make sure it doesn't shadow the cache variable.